src/data/finsupp/to_dfinsupp.lean

```diff
@@ -187,4 +192,63 @@ noncomputable def finsupp_lequiv_dfinsupp
   map_add' := finsupp.to_dfinsupp_add,
   .. finsupp_equiv_dfinsupp}
 
+section sigma
+/-- ### Stronger versions of `finsupp.split` -/
+
+noncomputable theory
+open_locale classical
+
+variables {η : ι → Type*} {N : Type*} [semiring R]
+
+open finsupp
+
+/-- `finsupp.split` is an equivalence between `(Σ i, η i) →₀ N` and `Π₀ i, (η i →₀ N)`. -/
+def sigma_finsupp_equiv_dfinsupp [has_zero N] : ((Σ i, η i) →₀ N) ≃ (Π₀ i, (η i →₀ N)) :=
+{ to_fun := λ f, ⟦⟨split f, (split_support f : finset ι).val, λ i,
+    begin
+    rw [← finset.mem_def, mem_split_support_iff_nonzero],
+    exact (decidable.em _).symm
+    end⟩⟧,
+  inv_fun := λ f,
+  begin
+    refine on_finset (finset.sigma f.support (λ j, (f j).support)) (λ ji, f ji.1 ji.2)
+      (λ g hg, finset.mem_sigma.mpr ⟨_, mem_support_iff.mpr hg⟩),
+    simp only [ne.def, dfinsupp.mem_support_to_fun],
+    intro h,
+    rw h at hg,
+    simpa using hg
+  end,
+  left_inv := λ f, by { ext, simp [split] },
+  right_inv := λ f, by { ext, simp [split] } }
+
+@[simp]
+lemma sigma_finsupp_equiv_dfinsupp_apply [has_zero N] (f : (Σ i, η i) →₀ N) :
+  (sigma_finsupp_equiv_dfinsupp f : Π i, (η i →₀ N)) = finsupp.split f := rfl
```

Comment:
I guess you should probably also add a lemma that `(sigma_finsupp_equiv_dfinsupp f).support = finsupp.split_support f`

Reply:
```lean
@[simp]
lemma sigma_finsupp_equiv_dfinsupp_support [has_zero N] (f : (Σ i, η i) →₀ N) :
  (sigma_finsupp_equiv_dfinsupp f).support = finsupp.split_support f :=
begin
  ext,
  rw dfinsupp.mem_support_to_fun,
  exact (finsupp.mem_split_support_iff_nonzero _ _).symm,
end
```

Reply:
Thank you!